### PR TITLE
Adds the notification timeline to mastodon.el closing #34.

### PR DESCRIPTION
**Pull request by alexjgriffith**
## Changes  
- Can now handle notifications (follow reblog favourite and mention)
- Tests for each of the rendering functions are included
- Added keybinding N to mastodon.el for mastodon-notifications-get
- added mastodon-notifications to ert-helper.el

## Note
This builds off of PR #174. Requesting a pull early to get feedback on commit f93d05c which adds the files `mastodon-notifications.el` and `mastodon-notifications-test.el` to the package.

### CODE REVIEW

**Comment by alexjgriffith** [comment]
It looks like the tests for rendering the warnings wont pass since they depend on libxml2.

**Comment by alexjgriffith** [comment]
Now there seams to be an issue with the linting. I will try to fix this on Friday.

File: lisp/mastodon-notifications.el [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+   (propertize message 'face 'highlight)
+   " "
+   (cdr (assoc message mastodon-notifications--response-alist))))
+
+(defun mastodon-notifications--mention (note)
+  "Format for a `mention' NOTE."
+  (let ((status (mastodon-tl--field 'status note)))
+    (mastodon-tl--insert-status
+     note
+     (replace-regexp-in-string
```
Comment (hdurer) [inline lisp/mastodon-notifications.el]: Might this be a good time to extract this regexp monster into a helper function somewhere?

File: lisp/mastodon-notifications.el [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+(defun mastodon-notifications--reblog (note)
+  "Format for a `boost' NOTE."
+  (let ((status (mastodon-tl--field 'status note)))
+    (mastodon-tl--insert-status
+     note
+     (replace-regexp-in-string
+      "[\t\n ]*\\'" ""
+      (if (mastodon-tl--has-spoiler status)
+          (mastodon-tl--spoiler status)
+                            (mastodon-tl--content status)))
```
Comment (hdurer) [inline lisp/mastodon-notifications.el]: Nit: indentation

File: test/mastodon-notifications-test.el [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+             (note . "E"))))
+
+(ert-deftest notification-get ()
+  "Ensure get request format for notifictions is accurate."
+  (let ((mastodon-instance-url "https://instance.url"))
+    (with-mock
+      (mock (mastodon-http--get-json "https://instance.url/api/v1/notifications"))
+      (mastodon-notifications--get))))
+
+(defun mastodon-notifications--test-type (fun sample)
```
Comment (hdurer) [inline test/mastodon-notifications-test.el:191]: Unused?

**Review by hdurer — COMMENTED** [review]
So cool!  I love it.

**Comment by hdurer** [comment]
BTW: If you wanted to write more tests but also got caught by the missing xml support:  
We can always put pre-rendered text (instead of HTML) into the json and just stub out the shr render call...

**Review by hdurer — APPROVED** [review]

**Pull request merged**